chemprop/train/loss_functions.py

```diff
@@ -363,3 +351,13 @@ def evidential_loss(pred_values, targets, lam=0, epsilon=1e-8):
     loss = L_NLL + lam * (L_REG - epsilon)
 
     return loss
+
+
+def quantile_loss_batch(pred_values, targets, quantiles):
+    """
+    Batched pinball loss at desired quantiles.
+    """
+
+    error = targets - pred_values
+
+    return torch.max((quantiles - 1) * error, quantiles * error)
```

Comment:
I don't understand this operation. Are you intending to use max to effectively do an absolute value?

What are the tensor shapes supposed to be? By my reading, targets and preds have a shape(batch, tasks) and quantiles has a shape(tasks). What did I miss where this isn't causing an error?

The convention for the other loss functions is to return a tensor of shape(batch), with a loss value for each datapoint in the batch. By using torch.max without setting an axis, you will get just one value. And the losses for all the datapoints but one will not have an effect on the gradient, which I'm assuming was not the intention.

Reply:
I'm using the broadcasting feature. preds, targets, error, results all have shape (batch, tasks) and quantiles has shape(tasks). The torch.max is between 2 different tensors (1 - quantiles) * error, -quantiles * error so it takes the element wise maximums of 2 tensors with shape (batch, tasks). I think the loss should be kept in the shape (batch, tasks) because its later summed in train.py line 179: loss = loss.sum() / mask.sum()